Falcon BMS Alternative Launcher/Override/OverrideSettingFor437.cs

```diff
@@ -22,5 +22,19 @@ protected override void OverrideVRHMD(StreamWriter cfg)
                 + Convert.ToInt32(mainWindow.Misc_VR.IsChecked)
                 + CommonConstants.CFGOVERRIDECOMMENT + "\r\n");
         }
+
+        protected override void WriteKeyLines(string filename, Hashtable inGameAxis, DeviceControl deviceControl, KeyFile keyFile, int DXnumber)
+        {
+            StreamWriter sw = new StreamWriter
+                (filename, false, Encoding.GetEncoding("utf-8"));
+            for (int i = 0; i < keyFile.keyAssign.Length; i++)
+                sw.Write(keyFile.keyAssign[i].GetKeyLine());
+            for (int i = 0; i < deviceControl.joyAssign.Length; i++)
+            {
+                sw.Write(deviceControl.joyAssign[i].GetKeyLineDX(i, deviceControl.joyAssign.Length, DXnumber));
+                sw.Write(deviceControl.joyAssign[i].GetKeyLinePOV());
```

Comment:
I haven't tested, but my worry is this will write out a block of pov-hat bindings for each HID joystick/gamepad device plugged in, in their DeviceSorting.txt order.  And I have no idea if BMS keyfile parser keeps the first 2, or the last 2..

Reply:
[BMS - Auto.key.txt](https://github.com/chihirobelmo/FalconBMS-Alternative-Launcher/files/12254756/BMS.-.Auto.key.txt)
You are right, it ended up outputting entries for my throttle, which does not have any POV hats, but it's setup XML contains (stub) 'PovAssgn' entries.

As far as I can tell, BMS is quite happy with this arrangement. I will to try plugging in a second joystick/device with POV hats and see whether that still holds up.